packages/mask/src/plugins/Savings/protocols/AAVEProtocol.ts

```diff
@@ -0,0 +1,358 @@
+import type Web3 from 'web3'
+import type { AbiItem } from 'web3-utils'
+import {
+    EthereumTokenType,
+    ChainId,
+    getSavingsConstants,
+    createContract,
+    FungibleTokenDetailed,
+    ZERO_ADDRESS,
+} from '@masknet/web3-shared-evm'
+import type { AaveLendingPool } from '@masknet/web3-contracts/types/AaveLendingPool'
+import type { AaveLendingPoolAddressProvider } from '@masknet/web3-contracts/types/AaveLendingPoolAddressProvider'
+
+import AaveLendingPoolAddressProviderABI from '@masknet/web3-contracts/abis/AaveLendingPoolAddressProvider.json'
+import AaveLendingPoolABI from '@masknet/web3-contracts/abis/AaveLendingPool.json'
+import BigNumber from 'bignumber.js'
+import { ProtocolCategory, SavingsNetwork, SavingsProtocol, ProtocolType } from '../types'
+import { pow10, ZERO } from '@masknet/web3-shared-base'
+
+export interface ContractListArray {
+    [index: string]: {
+        address: string
+    }
+}
+
+export interface AaveContract {
+    type: EthereumTokenType
+    chainName: string
+    subgraphUrl: string
+    aaveLendingPoolAddressProviderContract: string
+    aaveContract: string
+    stEthContract: string
+    assetContractAddresses: ContractListArray
+}
+
+export function getAaveContract(chainId: ChainId): AaveContract {
+    const constants = getSavingsConstants(chainId)
+
+    return {
+        type: EthereumTokenType.ERC20,
+        chainName: ChainId[chainId],
+        subgraphUrl: constants.AAVE_SUBGRAPHS ?? '',
+        aaveLendingPoolAddressProviderContract:
+            constants.AAVE_LENDING_POOL_ADDRESSES_PROVIDER_CONTRACT_ADDRESS ?? ZERO_ADDRESS,
+        aaveContract: constants.AAVE ?? ZERO_ADDRESS,
+        stEthContract: constants.AAVE ?? ZERO_ADDRESS,
+        assetContractAddresses: {
+            AAVE: { address: constants.AAVE ?? ZERO_ADDRESS },
+        },
+    }
+}
+
+export class AAVEProtocol implements SavingsProtocol {
+    public apr = '0.00'
+    public balance = ZERO
+    public readonly DEFAULT_APR = '0.17'
+
+    public availableNetworks: SavingsNetwork[] = []
+
+    public constructor(
+        public category = ProtocolCategory.ETH,
+        public type = ProtocolType.AAVE,
+        public name = 'AAVE',
+        public symbol = 'AAVE',
+        public image = 'aave',
+        public base: keyof Savings = 'AAVE',
```

Comment:
You should import the Savings JSON constants here.

Reply:
this throws an error

```
Class 'AAVEProtocol' incorrectly implements interface 'SavingsProtocol'.
  Types of property 'base' are incompatible.
    Type 'string | number | symbol' is not assignable to type 'string'.
      Type 'number' is not assignable to type 'string'
```